extensions/sql-database-projects/src/dialogs/deployDatabaseQuickpick.ts

```diff
@@ -70,6 +70,35 @@ export async function launchDeployAppIntegrationQuickpick(project: Project): Pro
 	};
 }
 
+async function launchEulaQuickPick(baseImage: string): Promise<boolean> {
+	const baseImages = uiUtils.getDockerBaseImages();
+	const imageInfo = baseImages.find(x => x.name === baseImage);
+	let result: string | undefined = constants.openEulaString;
+	if (imageInfo?.agreementInfo?.link) {
+		const link = imageInfo?.agreementInfo?.link;
+
+		// Ask user to update app settings or not
+		//
+		let choices = [
+			constants.yesString,
+			constants.noString,
+			constants.openEulaString
+		];
+		let options: vscode.QuickPickOptions = {
+			placeHolder: `${uiUtils.getAgreementDisplayText(imageInfo?.agreementInfo)}`,
+			ignoreFocusOut: true
+		};
+
+		while (result === constants.openEulaString) {
```

Comment:
Using buttons would be better (and remove the need for this), but as an FYI this would be better as a do...while

Reply:
can you show me an example of using buttons make it cleaner? so I won't need to use loop? the sample above didn't help that much

Reply:
or maybe I'm missing something. so if I check the result in onDidChangeSelection, I hide when I get "Yes" as result then I won't need a loop?

Reply:
I don't have a sample available - but you need to use createQuickPick and then set the buttons (which the sample linked has the code for, `input.buttons = ...`), hook up the callback (`onDidTriggerButton`) and then call `show()` on the quickpick. The rest of the logic should be the same.

Reply:
Cleaner from a UI perspective. Having an external action like this be part of the quick pick options isn't really intuitive (especially since most people aren't going to need to select it)

Reply:
that makes sense now. I thought you're talking about cleaner code. I made the change and using the folder icon for now but will ask for an icon